features/networking/egress-ip.feature

```diff
@@ -232,3 +232,127 @@ Feature: Egress IP related features
+    Given I store the masters in the :nodes clipboard
+    And the valid egress IP is added to the "<%= cb.nodes[0].name %>" node
+    And I have a project
+    And I have a pod-for-ping in the project
+
+    # add the egress ip to the project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
+
+    # Access some external host that is not responding
+    When I execute on the pod:
+      | nc | -v | -w2 | 8.8.8.8 | 23 |
+    Then the step should fail
+    And the output should contain "Operation timed out"
+
+    #Check the sdn logs on pod node
+    Given I use the "<%= pod.node_name %>" node
+    When I get the networking components logs of the node since "90s" ago
+    And the output should not contain "may be offline"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-25640
+  @admin
+  @destructive
+  Scenario: Should be able to access to the service's externalIP with egressIP
+    Given I have a project
+    Given I store the schedulable nodes in the :nodes clipboard
+    And the Internal IP of node "<%= cb.nodes[0].name %>" is stored in the :hostip clipboard
+
+    # Create additional network through CNO
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs":["<%= cb.hostip %>/24"]}}}} |
+
+    # Clean-up required to erase above externalIP policy after testing done
+    Given I register clean-up steps:
+    """
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs": null}}}} |
+    """
+
+    # Create a svc with externalIP
+    Given I switch to the first user
+    Given I obtain test data file "networking/externalip_service1.json"
+    And I wait up to 300 seconds for the steps to pass:
+    """
+    When I run oc create over "externalip_service1.json" replacing paths:
+      | ["spec"]["externalIPs"][0] | <%= cb.hostip %> |
+    Then the step should succeed
+    """
+
+    # Create a pod
+    Given I obtain test data file "routing/caddy-docker.json"
+    When I run the :create client command with:
+      | f | caddy-docker.json |
+    Then the step should succeed
+    And the pod named "caddy-docker" becomes ready
+
+    # Patch egressIP to the node
+    Given the valid egress IP is added to the node
+
+    # Patch egressIP to a new project
+    Given I create a new project
+    And I have a pod-for-ping in the project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
+
+    # Curl externalIP:portnumber from project patched egressIP should pass
+    When I execute on the pod:
+      | /usr/bin/curl | --connect-timeout | 10 | <%= cb.hostip %>:27017 |
+    Then the output should contain:
+      | Hello-OpenShift-1 http-8080 |
```

Comment:
egressIP testing in this case is trick, even curl passed here, but without sniffing the packets, you do not know the IP source address is from the added egressIP or nodeIP itself

Reply:
hmm, this case is from customer's issue : https://bugzilla.redhat.com/show_bug.cgi?id=1762580. The issue is with this configuration, the curl failed(It should use the egressIP, without egressIP configuration that means use the nodeIP, it succeeded).
When it configured with egressIP, and if the traffic access the node IP or  even the outside website, source IP should be translated to egressIP.  In 3.11, we installed the ipecho image on the node then curl $node_ip:port to get the IP source address. 
In 4.x auto, we cannot install the this image directly on the node, so use curl ifconfig.me to replace. we have other cases to check this point.
The logic never change.  Thanks!